Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -114,21 +114,27 @@ namespace AzToolsFramework
             // in focused instance DOM with the one seen by the root.
             else if (PrefabInstanceUtils::IsDescendantInstance(focusedInstance->get(), instance))
             {
-                // use instanceDom's allocator, because ultimately we will be setting this data back into
-                // instanceDom with move semantics.
-                PrefabDom focusedTemplateDomCopy(&instanceDom.GetAllocator());
+                // The template of the focused instance can be deleted in cases like hot reloading via file removal.
+                // So, check for a valid template id here before accessing its DOM
+                TemplateReference focusedTemplate= m_prefabSystemComponentInterface->FindTemplate(focusedInstance->get().GetTemplateId());
```

Comment:
You are correct. This doesn't happen anymore. So I'll remove the comment. I think it is still a valid check to have since other use cases in the future might run into this. I'll add an assert as this shouldn't happen.